src/WorkBC.ElasticSearch.Indexing/Services/XmlParsingServiceFederal.cs

```diff
@@ -374,9 +374,94 @@ public ElasticSearchJob ConvertToElasticJob(string federalXml, bool isFrench = f
                         }
                         else
                         {
-                            job.City = cacheLocation.City != null
-                                ? new[] { $"{VirtualJobBasedIn_EN} {cacheLocation.City}, {cacheLocation.Province}" }
-                                : new[] { $"{VirtualJobBasedIn_EN} {cacheLocation.Province}" };
+                            if (cacheLocation.City == null && cacheLocation.Province == null)
+                            {
+                                switch (employerPostalCode[..1])
+                                {
+                                    case "A":
+                                        job.City = isFrench ? new[] { $"{VirtualJobBasedIn_FR} Terre-Neuve-et-Labrador" }
+                                                            : new[] { $"{VirtualJobBasedIn_EN} Newfoundland and Labrador" };
+                                        break;
```

Comment:
Is Karim okay with us using `switch` statements or would he insist on a hash map?  I'm fine with it as is.

Reply:
Don't know, it hasn't come up. He saw the code when we discussed it, so I assume he's good with it or he would have said something then.